daemon/network.go

```diff
@@ -249,7 +267,9 @@ func (daemon *Daemon) createNetwork(create types.NetworkCreateRequest, id string
 	}
 	if nw != nil {
 		if create.CheckDuplicate {
-			return nil, libnetwork.NetworkNameError(create.Name)
+			if !agent || nw.Info().Dynamic() {
```

Comment:
In case of swarm networks, we have to allow creation even if a local network with the same name already exist.